packages/react-router-dom-v5-compat/lib/components.tsx

```diff
@@ -86,7 +86,7 @@ export function StaticRouter({
     pathname: locationProp.pathname || "/",
     search: locationProp.search || "",
     hash: locationProp.hash || "",
-    state: locationProp.state || null,
```

Comment:
@jungwoo3490 since you discovered this bug (🥳 ), I think this is not a full fix. You still can't set state to `""`, `false`, `undefined` or `NaN`. Maybe the change should be 

`state: locationProp.state`, which defaults to `undefined`, or with some check for other falsy values. What do you think.

Reply:
You're right. So I changed it to nullish coalescing operator and opened new PR.
Can you check my new PR?? https://github.com/remix-run/react-router/pull/11495

Reply:
Clever 👍